src/cljs/org/broadinstitute/firecloud_ui/common/components.cljs

```diff
@@ -211,21 +210,16 @@
            editing? (:editing? props)
            make-field
            (fn [entity key label dropdown? & [render]]
-               (if (and editing? dropdown?)
-                 [:div {}
-                  [:div {:style {}}
-                   [:span {:style {:fontWeight 500 :width 100 :display "inline-block" :paddingBottom "0.3em"}} label]
-                   (style/create-identity-select {:ref key
-                                                  :style {:width "100px"}
-                                                  :defaultValue (entity key)
-                                                  :onChange (when-let [f (:onSnapshotIdChange props)]
-                                                              #(f (int (common/get-text refs "snapshotId"))
-                                                                  (:namespace props)
-                                                                  (:name props)))}
-                                                 (:snapshots props))]]
-                 [:div {}
-                  [:span {:style {:fontWeight 500 :width 100 :display "inline-block" :paddingBottom "0.3em"}} label]
-                  [:span {} ((or render identity) (entity key))]]))
+             [:div {}
+              [:span {:style {:fontWeight 500 :width 100 :display "inline-block" :paddingBottom "0.3em"}} label]
+              (if (and editing? dropdown?)
+                (style/create-identity-select {:ref key
+                                               :style {:width "100px"}
+                                               :defaultValue (entity key)
+                                               :onChange (when-let [f (:onSnapshotIdChange props)]
+                                                           #(f (int (common/get-text refs "snapshotId"))))}
+                                              (:snapshots props))
+                [:span {} ((or render identity) (entity key))])])
```

Comment:
Factored the common stuff out.

I don't quite like the way this works--it seems like `:editing?` and `:onSnapshotIdChange` effectively duplicate each other.  But it's okay.